docs/index.md

```diff
@@ -1,23 +1,164 @@
+- [Examples](#examples) &mdash; concrete, end-to-end examples for common usage patterns
+- [More info](#more-info) &mdash; where to find support and more information
+
+## Usage guide
+
+The key primitives of TanStack DB are:
 
+- [collections](#defining-collections) &mdash; typed sets of objects that can be populated with data
+- [live queries](#using-live-queries) &mdash; to query data from/across collections and bind it to your components
+- [transactional mutators](#making-local-writes) &mdash; to apply local writes with background sync and managed optimistic state
 
-## Collections
+```tsx
+// Define collections to load data into
+const todoCollection = createCollection({
+  // your config
+})
+
+const Todos = () => {
+  // Bind data using live queries
+  const { data: todos } = useLiveQuery((query) =>
+    query
+      .from({ todoCollection })
+      .where('@completed', '=', false)
+  )
+
+  // Update data using transactional mutators
+  const updateTodo = useOptimisticMutation({ mutationFn })
+  const complete = (todo) =>
+    // Invokes the mutationFn to handle the write
+    updateTodo.mutate(() =>
+      // Instantly applies optimistic state
+      todoCollection.update(todo, (draft) => {
+        draft.completed = true
+      })
+    )
+
+  return (
+    <ul>
+      {todos.map(todo =>
+        <li key={ todo.id } onClick={() => complete(todo) }>
+          { todo.text }
+        </li>
+      )}
+    </ul>
+  )
+}
+```
+
+### Defining collections
+
+Collections are typed sets of objects that can be populated with data.
+
+They're designed to de-couple:
+
+- loading data into your app; from
+- binding data to your components
+
+Collections can be populated in many ways, including:
+
+- fetching data, for example [from API endpoints using TanStack Query](#)
+- syncing data, for example [using a sync engine like ElectricSQL](#)
+- storing local data, for example [in-memory client data or UI state](#)
+- from live queries, creating [derived collections as materialised views](#)
+
+Once you have your data in collections, you can query across them to bind results to your components. This allows you to:
+
+1. load normalised data into collections and then de-normalise it through queries; simplifying your backend by avoiding the need for bespoke API endpoints that match your client
+2. join data from multiple sources; for example, syncing some data out of a database, fetching some other data from an external API and then joining these into a unified data model for your front-end code
+
+Your components don't need to know where the data in the collections they're working with came from.
+
+### Using live queries
+
+Live queries are used to query data out of collections and bind the results to state variables in your components. Live queries are reactive: when the underlying data changes in a way that would affect the query result, the new result is automatically assigned to the state variable, triggering a re-render.
```

Comment:
not really sure what "bind the results to state variables"? Seems like you could drop the first sentence?